tests/com/jvm/external/jar/MergeJarsTest.java

```diff
@@ -351,6 +351,35 @@ public void aMergedJarShouldHaveTheManifestAsTheFirstOrSecondEntry() throws IOEx
     }
   }
 
+  @Test
+  public void shouldAddMissingDirectories() throws IOException {
+    Path input = temp.newFile("example.jar").toPath();
+    createJar(input, ImmutableMap.of("foo/bar/baz.txt", "Hello, World!"));
```

Comment:
Can you add a test case with both `foo/bar/baz.txt` and `foo/bar/baz/qux.txt` to make sure the directory creation log don't get confused at `baz`?

Reply:
Done!